### ISSUE TITLE
flutter example doesn't work

### ISSUE THREAD

**Issue by ConnorRigby** [braverhealth/phoenix-socket-dart#10]
```
E/flutter ( 7108): [ERROR:flutter/lib/ui/ui_dart_state.cc(177)] Unhandled Exception: NoSuchMethodError: The getter 'closeReason' was called on null.
E/flutter ( 7108): Receiver: null
E/flutter ( 7108): Tried calling: closeReason
E/flutter ( 7108): #0      Object.noSuchMethod  (dart:core-patch/object_patch.dart:51:5)
E/flutter ( 7108): #1      Null.closeReason  (dart:core/null.dart:1:1)
E/flutter ( 7108): #2      PhoenixSocket._onSocketClosed 
package:phoenix_socket/src/socket.dart:511
E/flutter ( 7108): #3      PhoenixSocket._onSocketError 
package:phoenix_socket/src/socket.dart:498
E/flutter ( 7108): #4      PhoenixSocket.connect 
package:phoenix_socket/src/socket.dart:200
E/flutter ( 7108): <asynchronous suspension>
E/flutter ( 7108):

```

**Comment by matehat** [comment]
@ConnorRigby do you have the example Phoenix Server running, from the example/backend directory?

**Comment by matehat** [comment]
I suspect the backend wasn't running, so that a connection error on socket startup, but it was badly handled. f3db04f
fixes that, and the fixes were published as part of the 0.4.6 release.
